### PR TITLE
Update CLDR to version 42

### PR DESCRIPTION
**Pull request by OzancanKaratas**
### Link to issue number:
Related `nvda-cldr` update: https://github.com/nvaccess/nvda-cldr/pull/3
### Summary of the issue:
Periodically CLDR update: CLDR team released version 42 of the CLDR package.
### Description of user facing changes
New emojis and characters
### Description of development approach
None
### Testing strategy:
Use your favorite emoji provider like emojipedia.org
### Known issues with pull request:
None
### Change log entries:
Changes
- Update CLDR to version 42. (#14273)
### Code Review Checklist:
- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -93,7 +93,7 @@ For reference, the following run time dependencies are included in Git submodule
 * [Sonic](https://github.com/waywardgeek/sonic), commit 1d705135
 * [IAccessible2](https://wiki.linuxfoundation.org/accessibility/iaccessible2/start), commit cbc1f29631780
 * [liblouis](http://www.liblouis.org/), version 3.23.0
-* [Unicode Common Locale Data Repository (CLDR)](http://cldr.unicode.org/), version 41.0
+* [Unicode Common Locale Data Repository (CLDR)](http://cldr.unicode.org/), version 42
```
Comment (seanbudd) [inline readme.md:96]: Probably should keep the `.0`

```suggestion
* [Unicode Common Locale Data Repository (CLDR)](http://cldr.unicode.org/), version 42.0
```

**Review by seanbudd — APPROVED** [review]

**Comment by leonardder** [comment]
I'm getting the following error on multiple systems when updating submodules:

```
fatal: transport 'file' not allowed                                                                                     
fatal: Fetched in submodule path 'nvda-cldr', but it did not contain 8e98c77fc3cdb33cf6d530f27f82c972ef3d93c0. Direct fetching of that commit failed.                                                                                           
```

**Comment by zstanecic** [comment]
Strange, when u have updated RHVoice language support, I was able to successfully update this.

**Comment by vielmetti** [comment]
This is almost certainly CVE-2022-39253 which prompted changes in git
due to security problems identified.

You may find this issue summary useful with an account of some workarounds
done by other projects to get around the "fatal: transport 'file' not allowed" problem.

https://vielmetti.typepad.com/logbook/2022/10/git-security-fixes-lead-to-fatal-transport-file-not-allowed-error-in-ci-systems-cve-2022-39253.html

**Pull request merged**